src/main/java/com/example/demo/client/JsonPlaceholderWebClient.java

```diff
@@ -0,0 +1,70 @@
+package com.example.demo.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+import java.util.Objects;
+
+@Component
+public class JsonPlaceholderWebClient {
```

Comment:
let's think how to generify our client to avoid code duplications in get single entity, or get a list of etities

Reply:
do not see todo methods

Reply:
let's have common 4 methods
- get with params (url, query, headers) which will return ClientResponse object (standard Spring class)
- post with params (url, query, headers, body) return ClientResponse
- put with params (url, query, headers, body) return ClientResponse
- patch with params (url, query, headers, body) return ClientResponse
- delete with params (url, query, headers, body) return ClientResponse
based on these methods you can define the methods which will return Post, list of Post, Todo, list of Todo

less methods not means simplicity